platform/pulpcore/app/status.py

```diff
@@ -0,0 +1,65 @@
+"""
+Module to return status information about a running Pulp instance
+"""
+from pkg_resources import get_distribution
+import logging
+
+from pulpcore.app.models.task import Worker
+from pulpcore.tasking.celery_instance import celery
+
+_logger = logging.getLogger(__name__)
+
+
+def get_version():
+    """
+    Gets the current version of the pulpcore package
+
+    Returns:
+        str: the current version of pulpcore
+    """
+    return get_distribution("pulpcore").version
+
+
+def get_db_conn_status():
+    """
+    Returns True if pulp is connected to the database
+
+    Returns:
+        bool: True if there's a db connection. False otherwise.
+    """
+    try:
+        Worker.objects.count()
+        return True
+    except:
+        return False
+
+
+def get_workers():
+    """
+    Gets a list of workers from the db
+
+    Returns:
+        list: a list of worker records
+    """
+    if get_db_conn_status():
+        workers = Worker.objects.all()
+        return workers
+    else:
+        return []
+
+
+def get_broker_conn_status():
+    """
+    Returns True if pulp can connect to the message broker
+
+    Returns:
+        bool: True if pulp can connect to the broker. False otherwise.
+    """
+    try:
+        conn = celery.connection()
+        conn.connect()
```

Comment:
The connection has a ``connected`` attribute. Would it be better to use that?

http://docs.celeryproject.org/projects/kombu/en/latest/reference/kombu.connection.html#kombu.connection.Connection.connected

If this function momentarily connects to the broker, but then disconnects, and the rest of Pulp is also disconnected from the broker, then this could be misleading.